File: Rust/String/Detect-Capital.rs
pub struct Solution;

impl Solution {
    pub fn detect_capital_use(word: String) -> bool {
        word.chars().skip(1).all(|x| x.is_lowercase()) || // For lowercase, first letter doesn't matter
                 word.chars().all(|x| x.is_uppercase())
    }
}
fn main() {}

File: Javascript/Detect-Capital.js
/**
 *
 * @param {string} word
 * @returns {boolean}
 */
const detectCapitalUse = (word) => {
  if (word.length == 1) {
    return true;
  }

  return (
    word
      .slice(1)
      .split("")
      .every((ch) => isLowercase(ch)) ||
    word.split("").every((ch) => isUppercase(ch))
  );
};

/**
 *
 * @param {string} word
 * @returns {boolean}
 */
const isLowercase = (word) => {
  return word.charCodeAt(0) >= 97 && word.charCodeAt(0) <= 122;
};

const isUppercase = (word) => {
  return word.charCodeAt(0) >= 65 && word.charCodeAt(0) <= 90;
};
